dbms/src/Flash/Statistics/ExchangeReceiverImpl.cpp

```diff
@@ -40,27 +42,30 @@ void ExchangeReceiverStatistics::appendExtraJson(FmtBuffer & fmt_buffer) const
     fmt_buffer.append("]");
 }
 
-void ExchangeReceiverStatistics::collectExtraRuntimeDetail()
+void ExchangeReceiverStatistics::updateExchangeReceiveDetail(const std::vector<ConnectionProfileInfo> & connection_profile_infos)
 {
-    const auto & io_stream_map = dag_context.getInBoundIOInputStreamsMap();
-    auto it = io_stream_map.find(executor_id);
-    if (it != io_stream_map.end())
+    RUNTIME_CHECK(connection_profile_infos.size() == partition_num);
+    for (size_t i = 0; i < partition_num; ++i)
     {
-        for (const auto & io_stream : it->second)
-        {
+        exchange_receive_details[i].packets += connection_profile_infos[i].packets;
+        exchange_receive_details[i].bytes += connection_profile_infos[i].bytes;
+    }
+}
+
+void ExchangeReceiverStatistics::collectExtraRuntimeDetail()
+{
+    if (tryTransformInBoundIOForStream(dag_context, executor_id, [&](const IBlockInputStream & stream) {
```

Comment:
Can we use more direct flag to check this?

Reply:
ok, added `ExecutionMode execution_mode` in `DAGContext` for judging execution mode